### ISSUE TITLE
MacOS Share ability

### ISSUE THREAD

**Issue by riartem** [telegramdesktop/tdesktop#1983]
Please, add supporting MacOS "share menu" with Telegram.
This menu used in all system apps and many side apps. 

![](http://i.imgur.com/pUpYq3t.png)

**Comment by AmeerTgh** [comment]
please add

**Comment by tdesktop-closer** [comment]
Hey there!

We're automatically closing this issue since there was no activity in this issue since 372 days ago. We therefore assume that the user has lost interest or resolved the problem on their own. Closed issues that remain inactive for a long period may get automatically locked.

Don't worry though; if this is in error, let us know with a comment and we'll be happy to reopen the issue.

Thanks!

<sub>(Please note that this is an [automated](https://github.com/auchri/github_auto_closer) comment.)</sub>

**Comment by EnricoMonese** [comment]
Any news on this?

**Comment by hutattedonmyarm** [comment]
Is this planned for the future? The old (now deprecated) Telegram for OSX version had this

**Comment by alexsergin** [comment]
+1

**Comment by ormwish** [comment]
+1

**Comment by sauron918** [comment]
it's actual

**Comment by stivens13** [comment]
Any progress?

**Comment by miickiie** [comment]
any update ?

**Comment by roberthchan** [comment]
+1

**Comment by sebastianarena** [comment]
+1000

**Comment by RememberTheAir** [comment]
Reactions are there for a reason. Please

**Comment by sebastianarena** [comment]
I just want to express how needed this is.
Given the fact that sharing on iOS is quite limited, since we need to keep the app open to share large files, having this would solve that immediately, as our macs are probably connected to power, and can stay open without restriction as a mobile phone.

The ability to report something is there for a reason, as well, didn't mean to be rude.

People starting asking for this 3+ years ago, it's quite a long time. Would definitely like to know if there's some way to move it up or know if its doable, per the reasons I stated above.

Thanks for a GREAT product, we love Telegram

**Comment by john-preston** [comment]
@sebastianarena I think you can use it in the native macOS Telegram app, https://macos.telegram.org/

**Comment by alexsergin** [comment]
@john-preston you can’t actually🙁

**Comment by overtake** [comment]
@alexsergin

share extension available on AppStore version of native macos client. 
https://apps.apple.com/us/app/telegram/id747648890

**Comment by stivens13** [comment]
Is there a specific reason why this feature cannot be implemented in TDesktop? If there was a statement, explaining why TDesktop cannot share files through mac menu, then all the discussion will be dropped

**Comment by john-preston** [comment]
@stivens13 I think somehow it can be, but it's hard to do.

**Comment by sebastianarena** [comment]
> @alexsergin
> 
> share extension available on AppStore version of native macos client.
> https://apps.apple.com/us/app/telegram/id747648890

Wait, really? I don't have the Mac Store version, I have the one that updates independently, and I don't see that... is it a Mac Store version exclusive feature??

**Comment by john-preston** [comment]
@sebastianarena Yes, this feature requires sandbox to be enabled. Enabling sandbox in your version will prevent auto-updates from working (in the non-appstore version).

**Comment by rizhenkov** [comment]
I think we can rename that feature to "Share Ability", because Windows has similar behavior.
Unigram already support this. Maybe tdesktop could so?
![изображение](https://user-images.githubusercontent.com/987517/65409850-79b0d680-ddf1-11e9-8eda-384f27d94993.png)

**Comment by GlassedSilver** [comment]
> 
> 
> @sebastianarena Yes, this feature requires sandbox to be enabled. Enabling sandbox in your version will prevent auto-updates from working (in the non-appstore version).

Cupertino at it again with their infinite wisdom.... *rolls eyes*

**Comment by khavran** [comment]
+1 for macos
